Bottom: 3945b1e8eac44042fcc1d080108d7fb91dbc2d8d
Top:    73f550397f2c1f73592ef3377d96bd8a8fdf2f94
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-03-23 00:10:46 +0000

Hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 8f023ac..58f8757 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -523,6 +523,15 @@ set_gl_color_for_gc (hidGC gc)
     return;
 
   free (priv->current_colorname);
+  priv->current_colorname = NULL;
+
+  /* If we can't set the GL colour right now, quit with
+   * current_colorname set to NULL, so we don't NOOP the
+   * next set_gl_color_for_gc call.
+   */
+  if (!priv->in_context)
+    return;
+
   priv->current_colorname = strdup (gtk_gc->colorname);
   priv->current_alpha_mult = gtk_gc->alpha_mult;
 
@@ -586,9 +595,6 @@ set_gl_color_for_gc (hidGC gc)
 #endif
   }
 
-  if(!priv->in_context)
-    return;
-
   hidgl_flush_triangles (gtk_gc->hidgl_gc.hidgl);
   glColor4d (r, g, b, a);
 }
@@ -884,6 +890,9 @@ draw_crosshair (hidGC gc, render_priv *priv)
   static int done_once = 0;
   static GdkColor cross_color;
 
+  if (!priv->in_context)
+    return;
+
   if (!done_once)
     {
       done_once = 1;
